### ISSUE TITLE
Feature request: Additional opt-out links defined by site admin

### ISSUE THREAD

**Issue by rupl** [appnexus/cmp#95]
I guess my question is two-fold:

1. Is there a technical solution to displaying additional opt-out links?
2. If it's technically feasible, would there be legal implications that would discourage a website owner from displaying non-IAB out-out links in this fashion?

I'm implementing this CMP for a client and it was requested that I add all opt-out links to the CMP itself within the allow/disallow/opt-out tables. It is desired that the links be present even if a vendor is not IAB-approved. An example is Google Analytics opt-out.

I tried manually modifying my generated `pubvendors.json` file with either a negative number, 0, or extremely large number as vendor ID. But it seems that if I add any vendor which isn't in the approved list, the tool defaults to showing the entire global vendor list instead of my reduced set.

**Comment by krishoyt** [comment]
There currently is no way to add in your own vendors if they are not part of the global list.  If you need that I would just manually add them to the interface.

I have no idea what the legal implications would be, but I imagine adding vendor IDs that are not globally known would make it hard for third parties to respect.

**Comment by rupl** [comment]
Thanks for the answer, and I agree it's probably best to avoid providing this functionality out of the box. I'm also unsure of the legal implications, but had a client asking that it be done that way. For anyone else who comes along curious if it's possible, here's what I did:

I copied the format of the vendor list and just merged them together during render to avoid polluting the store. Since they were opt-out links only, I avoided using `purposeIds` and only listed the appropriate `legIntPurposeIds`, which renders the extra vendors and their links.
